internal/crypto/certgen.go

```diff
@@ -70,11 +73,15 @@ const (
 // Certificates contains a set of Certificates as []byte each holding
 // the CA Cert along with Envoy Gateway & Envoy certificates.
 type Certificates struct {
-	CACertificate           []byte
-	EnvoyGatewayCertificate []byte
-	EnvoyGatewayPrivateKey  []byte
-	EnvoyCertificate        []byte
-	EnvoyPrivateKey         []byte
+	CACertificate             []byte
+	EnvoyGatewayCertificate   []byte
+	EnvoyGatewayPrivateKey    []byte
+	EnvoyCertificate          []byte
+	EnvoyPrivateKey           []byte
+	RateLimitCertificate      []byte
```

Comment:
curious why we need ratelimit and envoyratelimit, we just need one for the rl service xds client, right ?

Reply:
Yeah, we can reuse the existing cert, I have been undecided about this before, so I will change it.